src/timeMachine/reducers/index.ts

```diff
@@ -1081,6 +1098,87 @@ export const timeMachineReducer = (
         )
       })
     }
+
+    case SET_MAP_TYPE: {
+      const {mapType} = action
+
+      switch (mapType) {
+        case 'pointMap': {
+          return setViewProperties(state, {
+            layers: [defaultPointMap],
+          })
+        }
+        case 'heatmap': {
+          return setViewProperties(state, {
+            layers: [defaultHeatmap],
+          })
+        }
+        case 'trackMap': {
+          return setViewProperties(state, {
+            layers: [defaultTrackMap],
+          })
+        }
+        case 'circleMap': {
+          return setViewProperties(state, {
+            layers: [defaultCircleMap],
+          })
+        }
+        default:
+          return state
+      }
+    }
+    case SET_ZOOM_VALUE: {
+      const {zoom} = action
+      switch (state.view.properties.type) {
```

Comment:
code style tip: no need to get fancy here, just use a normal early return guard statement:

```js
if (state?.view?.properties?.type === 'geo') {
  return setViewProperties(state, {zoom})
}

return state
}
```

**BUT** is this even necessary? Won't at this point, the `type` always be geo?